### PR TITLE
#1024 A few bug fixes I'd like to add to v2.1 series (v2.1.5)

### PR DESCRIPTION
**Pull request by rmjarvis**
This PR includes a few bug fixes I think we should include in the 2.1 series as v2.1.5:

1. @aamon pointed out a bug in the WCS functions where things like `local` would fail when given a PositionI input rather than a PositionD.  This is a casualty of a few-years-old change in numpy behavior to disallow subtracting a float from an int with `-=` (and similar ops).  But apparently it's rare enough that no one noticed it being a problem until now.  Anyway, the fix is just to build most of our numpy arrays with `dtype=float`.  I went through and checked them all and fixed the ones that looked potentially problematic, but I only added a unit test for the error that Alex was getting.
2. PyYAML changed its `load` function to raise ugly errors telling people to switch to `safe_load` instead or explicitly use a non-safe `Loader`.  I'm pretty sure we don't ever need a non-safe loader, so I switched to `safe_load`.
3. I fixed the inefficient decompression step that gets triggered by `FitsHeader`, even though it doesn't actually need the data, just the header info.  (cf. #1024, reported by @esheldon)
4. I cherry-picked @rmandelb's fix to #1020, since I thought we might as well include that in v2.1.5 as well.  (Ping @beckermr)\
5. I fixed the seed checks for BaseDeviate to allow non-int integers (cf. #1009, reported by @matroxel).

Note: This is set to merge to releases/2.1.  I will cherry-pick them over to master by hand.

### CODE REVIEW

File: tests/test_random.py [hunk 1]
```diff
@@ -1683,6 +1683,40 @@ def test_ne():
...
+              np.uint32(123),
+              np.uint64(123),
+              np.short(123),
+              np.ushort(123),
+              np.intc(123),
+              np.uintc(123),
+              np.intp(123),
+              np.uintp(123),
+              np.int_(123),
+              np.longlong(123),
```
Comment (jmeyers314) [inline tests/test_random.py]: I suppose `np.long` and `np.ulong` should be added for completeness.

Comment (rmjarvis) [reply tests/test_random.py]: I think I got all the numpy valid types.  It's true that 
`np.long` is valid (albeit [undocumented](https://docs.scipy.org/doc/numpy/user/basics.types.html)).  But it is equivalent to `int` in Python 3.x and `long` in Python 2.7, so not really an additional type to check.  

However, I don't think `np.ulong` is valid.
```
>>> np.long
<class 'int'>
>>> np.ulong
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'numpy' has no attribute 'ulong'
```

Comment (jmeyers314) [reply tests/test_random.py]: Oh interesting.  I think I somehow ended up at the [v1.10 documentation](https://docs.scipy.org/doc/numpy-1.10.1/user/basics.types.html) which implies that long and ulong are a thing.  But I guess this doesn't exist in recent numpy.

**Comment by rmjarvis** [comment]
Done.  v2.1.5 released.

**Pull request merged**